cogs/StaffAppsBackoffice.py

```diff
@@ -0,0 +1,966 @@
+        self.author = author
+        self.data = data
+        self.cur_page = 1
+        self.max_page = len(data) // 10 + 1
+
+    async def repopulate(self, interaction: discord.Interaction) -> None:
+        """
+        Repopulates the view with the next page of data after a button press.
+        """
+        title = f"Applications page {self.cur_page}/{self.max_page}"
+        field_title = "Application ID, Discord Name, Status, Submission Time"
+        field_content = ""
+        for row in self.data[self.cur_page * 10 - 10 : self.cur_page * 10]:
+            field_content += (
+                f"{row[0]}, {row[1]}, {row[2]}, {datetime.fromtimestamp(int(row[3]))}\n"
+            )
+        fields = [[field_title, field_content, False]]
+        embed = EmbedBuilder(
+            title=title,
+            description="",
+            fields=fields,
+            color=0x24AAE1,
+        )  # hwh blue
+        await interaction.response.edit_message(embed=embed.build(), view=self)
+
+    @discord.ui.button(
+        label="",
+        style=discord.ButtonStyle.primary,
+        emoji="⬅️",
+        disabled=True,
+    )
+    async def button_callback(
+        self,
+        button: discord.ui.Button,
+        interaction: discord.Interaction,
+    ) -> None:
+        self.children[1].disabled = False
+        self.cur_page -= 1
+        if self.cur_page == 1:
+            button.disabled = True
+        await self.repopulate(interaction)
+
+    @discord.ui.button(
+        label="",
+        style=discord.ButtonStyle.primary,
+        emoji="➡️",
+        disabled=False,
+    )
+    async def button2_callback(
+        self,
+        button: discord.ui.Button,
+        interaction: discord.Interaction,
+    ) -> None:
+        self.children[0].disabled = False
+        self.cur_page += 1
+        if self.cur_page == self.max_page:
+            button.disabled = True
+        await self.repopulate(interaction)
+
+
+class staffAppsSeeDeniedAccepted(discord.ui.View):
+    """
+    A discord view for the denied and accepted commands
+    """
+
+    def __init__(self, author: discord.Member, data: list, da: str) -> None:
+        super().__init__()
+        self.author = author
+        self.data = data
+        self.cur_page = 1
+        self.max_page = len(data) // 10 + 1
+        self.da = da  # d for denied, a for accepted
+
+    async def repopulate(self, interaction: discord.Interaction) -> None:
+        """
+        Repopulates the view with the next page of data after a button press.
+        """
+        title = f"{'Denied' if self.da == 'd' else 'Accepted'} Applications page {self.cur_page}/{self.max_page}"
+        field_title = "Application ID, Discord Name, Original Submission Time"
+        field_content = ""
+        for row in self.data[self.cur_page * 10 - 10 : self.cur_page * 10]:
+            field_content += (
+                f"{row[0]}, {row[1]}, {datetime.fromtimestamp(int(row[2]))}\n"
+            )
```

Comment:
Thoughts on using `.join()`?
```suggestion
        field_content = "".join(
            f"{row[0]}, {row[1]}, {datetime.fromtimestamp(int(row[2]))}\n"
            for row in self.data[self.cur_page * 10 - 10 : self.cur_page * 10]
        )
```

Reply:
good idea